module/motion/IKKick/src/IKKick.cpp

```diff
@@ -132,26 +128,28 @@ namespace motion {
                     supportFoot = LimbID::RIGHT_LEG;
                 }
 
-                Transform3D torsoPose = (supportFoot == LimbID::LEFT_LEG) ? leftFoot.i() : rightFoot.i();
+                Eigen::Affine3d torsoPose =
+                    (supportFoot == LimbID::LEFT_LEG) ? leftFoot.inverse() : rightFoot.inverse();
 
                 // Put the ball position from vision into torso coordinates
-                arma::vec3 targetTorso;  // = Transform3D(convert<double, 4,
-                                         // 4>(sensors.kinematicsBodyToGround)).i().transformPoint(convert<double,
-                                         // 3>(command.target)); //TODO fix
+                Eigen::Vector3d targetTorso;  // = Transform3D(convert<double, 4,
+                                              // 4>(sensors.kinematicsBodyToGround)).i().transformPoint(convert<double,
+                                              // 3>(command.target)); //TODO fix
                 // Put the ball position into support foot coordinates
-                arma::vec3 targetSupportFoot = torsoPose.transformPoint(targetTorso);
+                Eigen::Vector3d targetSupportFoot = torsoPose * targetTorso;
 
                 // Put the goal from vision into torso coordinates
-                arma::vec3 directionTorso;  // = Transform3D(convert<double, 4,
-                                            // 4>(sensors.kinematicsBodyToGround)).i().transformVector(convert<double,
-                                            // 3>(command.direction)); //TODO fix
+                Eigen::Vector3d
+                    directionTorso;  // = Transform3D(convert<double, 4,
+                                     // 4>(sensors.kinematicsBodyToGround)).i().transformVector(convert<double,
+                                     // 3>(command.direction)); //TODO fix
```

Comment:
What is stopping you from fixing these?

Reply:
Are these your `TODO fix` comments?

Reply:
Nope they aren't mine. No idea what they mean

Reply:
@TrentHouliston do you know anything about these `TODO`s?

Reply:
I think we need to do something with this vector. The following code makes no sense without this.